Bluetooth/Bluetooth.cpp

```diff
@@ -653,6 +662,83 @@ namespace WPEFramework
+             }
+             else if (Utils::String::contains(deviceProfile, "SMARTPHONE") ||
+                      Utils::String::contains(deviceProfile, "TABLET")) {
+                      lenDevOpDiscType = BTRMGR_DEVICE_OP_TYPE_AUDIO_INPUT;
+             }
+             else if (Utils::String::contains(deviceProfile, "KEYBOARD") ||
+                      Utils::String::contains(deviceProfile, "MOUSE") ||
+                      Utils::String::contains(deviceProfile, "JOYSTICK")) {
+                      lenDevOpDiscType = BTRMGR_DEVICE_OP_TYPE_HID;
+             }
+             else if (Utils::String::contains(deviceProfile, "LE TILE") ||
+                      Utils::String::contains(deviceProfile, "LE")) {
+                      lenDevOpDiscType = BTRMGR_DEVICE_OP_TYPE_LE;
+             }
+             else if (Utils::String::contains(deviceProfile, "DEFAULT")) {
+                      lenDevOpDiscType = BTRMGR_DEVICE_OP_TYPE_UNKNOWN;
+             }
+
+             rc = BTRMGR_SetDeviceVolumeMute (0, deviceHandle, lenDevOpDiscType, ui8volume, mute);
+             return BTRMGR_RESULT_SUCCESS == rc;
+        }
+
+        JsonObject Bluetooth::getDeviceVolumeMuteProperties(long long int  deviceID, const string &deviceProfile)
+        {
+             BTRMGR_Result_t rc = BTRMGR_RESULT_SUCCESS;
+             BTRMgrDeviceHandle deviceHandle = (BTRMgrDeviceHandle) deviceID;
+             BTRMGR_DeviceOperationType_t lenDevOpDiscType = BTRMGR_DEVICE_OP_TYPE_AUDIO_OUTPUT;
+	     unsigned char ui8volume;
+	     unsigned char mute;
+	     JsonObject volumeInfo;
+
+             if (Utils::String::contains(deviceProfile, "LOUDSPEAKER") ||
+                 Utils::String::contains(deviceProfile, "HEADPHONES") ||
+                 Utils::String::contains(deviceProfile, "WEARABLE HEADSET") ||
+                 Utils::String::contains(deviceProfile, "HIFI AUDIO DEVICE")) {
+                 lenDevOpDiscType = BTRMGR_DEVICE_OP_TYPE_AUDIO_OUTPUT;
+             }
+             else if (Utils::String::contains(deviceProfile, "SMARTPHONE") ||
+                      Utils::String::contains(deviceProfile, "TABLET")) {
+                      lenDevOpDiscType = BTRMGR_DEVICE_OP_TYPE_AUDIO_INPUT;
+             }
+             else if (Utils::String::contains(deviceProfile, "KEYBOARD") ||
+                      Utils::String::contains(deviceProfile, "MOUSE") ||
+                      Utils::String::contains(deviceProfile, "JOYSTICK")) {
+                      lenDevOpDiscType = BTRMGR_DEVICE_OP_TYPE_HID;
+             }
+             else if (Utils::String::contains(deviceProfile, "LE TILE") ||
+                      Utils::String::contains(deviceProfile, "LE")) {
+                      lenDevOpDiscType = BTRMGR_DEVICE_OP_TYPE_LE;
+             }
+             else if (Utils::String::contains(deviceProfile, "DEFAULT")) {
+                      lenDevOpDiscType = BTRMGR_DEVICE_OP_TYPE_UNKNOWN;
+             }
```

Comment:
Since this code is repeated, you should create a static function in this .cpp like BTRMGR_DeviceOperationType_FromString(const std::string & s);

Reply:
thanks for review . Addressed in latest diffs.